src/function/maybe_changed_after.rs

```diff
@@ -452,12 +456,10 @@ where
                             inputs,
                         );
 
-                        if is_provisional {
```

Comment:
I guess it's cheaper to unconditionally update this, even when it's not necessary?

Reply:
I guess I could revert this now. I first split `deep_verify_memo` into two functions so that I didn't had access to `is_provisional` anymore. I'll revert to reduce the size of the change.